relay-server/src/services/spooler/envelope_stack/sqlite.rs

```diff
@@ -259,34 +288,37 @@ impl EnvelopeStack for SQLiteEnvelopeStack {
         }
 
         self.batches_buffer_size += 1;
-
-        Ok(())
     }
 
-    async fn peek(&mut self) -> Result<&Box<Envelope>, Self::Error> {
+    async fn peek(&mut self) -> Result<Option<&Box<Envelope>>, Self::Error> {
+        self.try_wait_spooling().await?;
+
         if self.below_unspool_threshold() && self.check_disk {
             self.unspool_from_disk().await?
         }
 
-        self.batches_buffer
+        let last = self
+            .batches_buffer
             .back()
-            .and_then(|last_batch| last_batch.last())
-            .ok_or(Self::Error::Empty)
+            .and_then(|last_batch| last_batch.last());
```

Comment:
Nit: `&Box<Envelope>` is a double indirection. If you want you can turn it into `&Envelope` by doing
```suggestion
            .and_then(|last_batch| last_batch.last().map(|e| e.as_ref()));
```

Reply:
This is a good point, I decided for `&Box` for uniformity of the trait but I will change it since I agree with your comment. Thanks!